grpc/src/status/server_status.rs

```diff
@@ -0,0 +1,67 @@
+use super::status_code::StatusCode;
+use super::Status;
+
+/// Represents a gRPC status on the server.
+///
+/// This is a separate type from `Status` to prevent accidental conversion and
+/// leaking of sensitive information from the server to the client.
+#[derive(Debug, Clone)]
+pub struct ServerStatus(Status);
+
+impl std::ops::Deref for ServerStatus {
+    type Target = Status;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl ServerStatus {
+    /// Create a new `ServerStatus` with the given code and message.
+    pub fn new(code: StatusCode, message: impl Into<String>) -> Self {
+        ServerStatus(Status::new(code, message))
+    }
+
+    /// Converts the `ServerStatus` to a `Status` for client responses.
+    pub(crate) fn to_status(self) -> Status {
```

Comment:
Per the clippy warning this probably wants to be named `into_status(self)` instead.